### PR TITLE
WIP: Feature/fragment cache

### PR DESCRIPTION
**Pull request by bricesanchez**

### CODE REVIEW

File: core/app/views/refinery/_header.html.erb [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <% end %>
 <% end %>
 
-<% cache [I18n.locale, 'main_menu', @page, Refinery::Page.fast_menu.map(&:updated_at).max] do %>
+<% cache [I18n.locale, current_refinery_user, @page, Refinery::Page.fast_menu] do %>
```
Comment (parndt) [inline core/app/views/refinery/_header.html.erb]: huh? There is no `current_refinery_user` for this frontend view

Comment (bricesanchez) [reply core/app/views/refinery/_header.html.erb]: Yes there is a `current_refinery_user`: https://github.com/refinery/refinerycms/blob/2c3b7c31314cc3fbcd425d7c461e14002bfda7fc/core/lib/refinery/application_controller.rb#L51

Comment (parndt) [reply core/app/views/refinery/_header.html.erb]: but it's going to be nil when you're logged out so your cache key will be `en__#{@page}_#{array of pages}` which seems like a strange key. What am I missing?

Comment (bricesanchez) [reply core/app/views/refinery/_header.html.erb]: Probably, I will change the way i generate the cache_key, i will also using the model `cache_key` method. But right now i've a `can't dump File` error in the specs and i don't know why.

File: pages/app/presenters/refinery/pages/content_page_presenter.rb [hunk 2]
```diff
@@ -5,17 +5,22 @@ module Pages
...
         add_page_parts(page.parts) if page
         add_default_post_page_sections
       end
 
-    private
+      private
 
       def add_default_title_section(title)
-        add_section TitleSectionPresenter.new(:id => :body_content_title, :fallback_html => title)
+        Rails.cache.fetch([::I18n.locale, @current_refinery_user, @page, title]) do
```
Comment (parndt) [inline pages/app/presenters/refinery/pages/content_page_presenter.rb]: previously these presenters had no association with `Rails`, nor to users or anything like that. I'm uncomfortable adding those when caching can happen in the layer that has these.

Comment (bricesanchez) [reply pages/app/presenters/refinery/pages/content_page_presenter.rb]: I understand, how should we handle this?

**Comment by parndt** [comment]
Is it something we need to worry about / maintain?

File: pages/app/presenters/refinery/pages/content_page_presenter.rb (hunk 2 above)
Comment (parndt) [reply pages/app/presenters/refinery/pages/content_page_presenter.rb]: Is it something we need to worry about / maintain?

Comment (bricesanchez) [reply pages/app/presenters/refinery/pages/content_page_presenter.rb]: I think you're right, we should not worry about it right now:
* cache_pages_full_caching exists
* when a template is more complex, we don't fully use the `ContentPagePresenter`, so we can easily add Rais fragment cache per project.

After digging into this behavior, i think it's probably the admin pages#index who need a better cache: https://github.com/refinery/refinerycms/blob/master/pages/app/views/refinery/admin/pages/_page.html.erb

**Comment by parndt** [comment]
Thanks Brice

**Pull request closed without merging**